webui/react/src/pages/ExperimentList.tsx

```diff
@@ -162,12 +162,14 @@ const ExperimentList: React.FC = () => {
 
     // archived
     const archived = urlSearchParams.get('archived');
-    filters.archived = archived === URL_ALL ? undefined : archived as ArchiveFilter;
+    if (archived != null) {
+      filters.archived = archived === URL_ALL ? undefined : archived as ArchiveFilter;
+    }
```

Comment:
this is the real fix, the other changes are just slight improvement refactoring.